[![](imagens/colab-badge.png){width="16%"}](https://colab.research.google.com/github/fzampirolli/pdi-vc/blob/master/notebooks_alunos/apendices/apendice_f_aluno.ipynb)
[![](imagens/github-badge.png){width="19%"}](https://github.com/fzampirolli/pdi-vc)

# Análise de Percepção e Avaliação Didática: PDI-VC 2026.2 {#apendice-pdi-vc-2026-2}

Este apêndice apresenta uma análise **quantitativa e qualitativa, agregada e não identificável**, da percepção dos estudantes sobre a experiência pedagógica na disciplina de **Processamento Digital de Imagens e Visão Computacional (PDI-VC)**, oferecida no segundo quadrimestre de 2026. A apresentação dos resultados foi estruturada de modo a preservar a **confidencialidade e a não identificação dos participantes**, não sendo divulgados nomes, endereços eletrônicos, respostas individualizadas ou quaisquer outros elementos que permitam a identificação direta ou indireta de estudantes.

::: {.callout-note}

## Proteção da confidencialidade, tratamento dos dados e reprodutibilidade

As análises apresentadas neste apêndice são realizadas exclusivamente sobre **dados agregados ou anonimizados**, utilizados para fins de avaliação e análise da experiência pedagógica. Os resultados são apresentados de forma coletiva, sem associação entre respostas e identidades individuais.

As tabelas, estatísticas e gráficos são gerados de forma **automática e reprodutível** a partir dos dados utilizados na análise. O *notebook Colab* disponibilizado pelo ícone no canto superior esquerdo documenta os procedimentos computacionais empregados, incluindo o processamento, a transformação e a geração dos resultados apresentados neste apêndice.

Os **dados brutos contendo informações potencialmente identificáveis não são publicados**, incorporados ao material de divulgação ou disponibilizados por meio do *notebook*. Da mesma forma, respostas individuais não são apresentadas de maneira que possibilite sua atribuição a participantes específicos. Eventuais materiais necessários à **auditoria metodológica** permanecem sob guarda do responsável pela pesquisa e poderão ser disponibilizados, quando cabível, exclusivamente em condições compatíveis com as exigências éticas aplicáveis e sem exposição de dados pessoais ou identificáveis.

A identificação utilizada exclusivamente para o pareamento entre os questionários PRÉ e PÓS, quando necessária à análise longitudinal, não é divulgada nos resultados. Após a realização dos procedimentos analíticos que dependem desse pareamento, os resultados são apresentados de forma agregada, preservando-se a confidencialidade dos participantes.

:::


## Caracterização da Amostra e Metodologia de Coleta

A coleta de dados ocorreu em dois momentos do quadrimestre. O **questionário PRÉ-curso**, aplicado na primeira semana de aula, caracterizou o perfil dos estudantes antes da intervenção pedagógica. O **questionário PÓS-curso**, aplicado ao final do quadrimestre, após a utilização dos recursos didáticos e das estratégias analisadas neste apêndice, avaliou a experiência na disciplina e foi aplicado juntamente com a prova final.

Foram obtidas **104 respostas no PRÉ** e **80 no PÓS**. Com base nas informações fornecidas pelos estudantes para possibilitar a identificação entre os dois momentos, foram identificados **15 estudantes que responderam a ambos os questionários**, constituindo a **amostra pareada** utilizada na análise da evolução individual.

A distribuição dos respondentes entre as três turmas é apresentada na @tbl-contagem-turmas. As turmas **DA1** e **DA2** correspondem à graduação, oferecida no período **matutino**, enquanto a **MCC** corresponde à pós-graduação, oferecida no período **vespertino**. Entre o PRÉ e o PÓS, o número de respostas **manteve-se em 39 na DA1**, enquanto diminuiu de **46 para 30 na DA2** e de **19 para 11 na MCC**. Assim, a redução da amostra concentrou-se nas turmas **DA2 e MCC**, enquanto a participação da **DA1 permaneceu estável**.


Considerando a natureza ordinal das escalas *Likert* de cinco pontos, foram empregados **testes estatísticos não paramétricos**. O teste de *Mann--Whitney U* [@mann1947] foi utilizado nas comparações entre amostras independentes; o teste de *Kruskal--Wallis* [@kruskal1952], nas comparações entre mais de dois grupos independentes; e o teste de Friedman [@friedman1937], na comparação entre diferentes recursos avaliados pelos mesmos estudantes. Para a comparação pareada entre PRÉ e PÓS, empregou-se o teste de *Wilcoxon* [@wilcoxon1945].

Como complemento aos testes de significância, foram analisados **tamanhos de efeito**, **correlações de *Spearman***, **consistência interna pelo $\alpha$ de *Cronbach*** e **preferências declaradas**, permitindo uma interpretação mais ampla dos resultados.

::: {.callout-note}

### Nota Metodológica {.unnumbered}

Os testes estatísticos permitem identificar diferenças e associações, mas não estabelecem relações causais. Assim, diferenças entre recursos que utilizaram ou não IA não devem ser interpretadas isoladamente como efeito da tecnologia, uma vez que esses recursos apresentam finalidades, formatos, níveis de interação e momentos de utilização distintos.
:::

In [22]:
#| echo: false     # esconde o código, mas mantém o output (tabela/gráfico)
#| include: false  # esconde código E output, mas ainda executa (útil pra imports/setup)
#| output: false   # mostra o código, mas esconde o output

import os
import re
import matplotlib.lines as mlines
import matplotlib.patches as mpatches
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from scipy import stats
import seaborn as sns

pd.set_option("display.max_columns", 200)
pd.set_option("display.width", 200)

# --------------------------------------------------------------------------
# CONFIGURAÇÃO: caminhos dos arquivos CSV locais na pasta dataset/
# --------------------------------------------------------------------------
PATH_POS = os.path.join("dataset", "POS_anonimizado.csv")
PATH_PRE = os.path.join("dataset", "PRE_anonimizado.csv")

# ==============================================================================
# PADRÃO VISUAL DO NOTEBOOK — usado em TODOS os gráficos a partir daqui.
# Qualquer ajuste de cor/estilo deve ser feito aqui, uma única vez.
# ==============================================================================
CORES = {
    "pre":            "#91bfdb",   # respostas do momento PRÉ
    "pos":            "#a6d96a",   # respostas do momento PÓS 
    "aprovado":       "#2b5c8f",   # item significativo / aprovado (azul escuro) 
    "neutro":         "#8c96c6",   # item não significativo (roxo acinzentado) 
    "linha_neutra":   "#d73027",   # linha de referência do ponto neutro (3,0) 
    "media_marker":   "#d73027",   # marcador de média em boxplots 
    "outlier":        "#e67e22",   # destaque de casos excluídos/atípicos 
    "likert_5cores":  ["#d7191c", "#fdae61", "#ffffbf", "#a6d96a", "#1a9641"],  # notas 1→5 
    "efeito_pequeno": "#3498db", 
    "efeito_medio":   "#e67e22", 
    "efeito_grande":  "#27ae60", 
    "efeito_nulo":    "#7f8c8d", 
    "heatmap_cmap":   "YlGnBu_r", 
    "texto_escuro":   "#111111", 
    "texto_medio":    "#333333", 
}
FONTE = {"titulo": 13, "subtitulo": 12, "eixo": 11, "texto": 9.5, "legenda": 9} 

sns.set_theme(style="whitegrid", context="notebook", palette="muted") 
plt.rcParams["figure.figsize"] = (10, 6) 
plt.rcParams["font.size"] = 10 
plt.rcParams["axes.titleweight"] = "bold" 

def extrair_descricao(col, max_len=90):
    """Extrai prioritariamente o texto contido dentro de colchetes [...] do cabeçalho do Forms.
    Se não houver colchetes com texto descritivo, limpa o texto do enunciado."""
    # Procura por conteúdos dentro de colchetes [...]
    m = re.findall(r'\[(.*?)\]', str(col))
    if m:
        # Filtra códigos simples como [Q08_02] para pegar o texto/rótulo completo do colchete
        # Exemplo: de "[Q09_02 - Simuladores]" pega "Q09_02 - Simuladores"
        textos = [t.strip() for t in m if len(t.strip()) > 0]
        if textos:
            # Seleciona o trecho entre colchetes mais informativo (o maior ou mais descritivo)
            desc = sorted(textos, key=len, reverse=True)[0]
            if len(desc) > max_len:
                desc = desc[:max_len - 1].rstrip() + "…"
            return desc

    # Fallback (caso não haja colchetes na coluna)
    desc = re.sub(r"\[+(PRE_Q\d+|Q\d+(?:_\d+)?)\]*", "", str(col)).strip(" -–:\u200b\"")
    desc = re.sub(r"^\d{1,2}[\.\)]\s*", "", desc)
    desc = re.sub(r"\s+", " ", desc).strip()
    if len(desc) > max_len:
        desc = desc[:max_len - 1].rstrip() + "…"
    return desc

def extrair_codigo(col):
    """Extrai um código curto tipo Q01, Q08_05, PRE_Q06 do cabeçalho longo do Forms."""
    m = re.search(r"\[+(PRE_Q\d+|Q\d+(?:_\d+)?)", col)
    if m:
        return m.group(1)
    if col.strip() == "" or col.startswith("[Q09"):
        return None  # coluna de checkbox sem rótulo (bug identificado) -> tratada à parte
    return col  # mantém original (ex.: Carimbo, Turma, e-mail)

def estilo_eixos(ax):
    """Acabamento padrão de todos os gráficos: remove as bordas superior e direita.""" 
    ax.spines["top"].set_visible(False) 
    ax.spines["right"].set_visible(False) 
    return ax 


def linha_neutra(ax, valor=3.0, orientacao="v", label="Ponto Neutro (3,0)"):
    """Desenha a linha de referência do ponto neutro da escala, com o estilo padrão.""" 
    if orientacao == "v": 
        ax.axvline(valor, color=CORES["linha_neutra"], linestyle="--", linewidth=1.5, alpha=0.85, label=label) 
    else:
        ax.axhline(valor, color=CORES["linha_neutra"], linestyle="--", linewidth=1.5, alpha=0.85, label=label) 

In [23]:
#| label: setup
#| echo: false
#| include: false

# --------------------------------------------------------------------------
# Opções de chunk usadas aqui:
#   echo: false     -> esconde o código, mantém o output (tabela/gráfico)
#   include: false  -> esconde código E output, mas ainda executa (setup/imports)
#   output: false   -> mostra o código, esconde só o output
# --------------------------------------------------------------------------

import json
import os

# ==============================================================================
# 1. CARREGAMENTO DOS DADOS
# ==============================================================================
def carregar_pipeline():
    """Carrega PRÉ e PÓS anonimizados do zero e normaliza a coluna Turma
    (funde grafias divergentes como 'CCM'/'MCC')."""

    def _carregar(caminho_csv):
        df = pd.read_csv(caminho_csv)
        df.columns = [str(c).strip() for c in df.columns]
        return df

    def _normalizar_turmas(df, coluna="Turma", mapeamento=None):
        """Normaliza grafias divergentes da coluna Turma (ex.: 'CCM' e 'MCC' são a
        mesma turma, só com a sigla do curso invertida). Aplica strip + upper antes
        de mapear, para pegar também variações de espaço/caixa (' mcc', 'Mcc' etc.)."""
        if mapeamento is None:
            mapeamento = {"CCM": "MCC"}  # ajuste aqui se o canônico deveria ser CCM
        df = df.copy()
        turma_normalizada = df[coluna].astype(str).str.strip().str.upper()
        df[coluna] = turma_normalizada.replace(mapeamento)
        return df

    pos = _carregar(PATH_POS)
    pre = _carregar(PATH_PRE)
    pos = _normalizar_turmas(pos)
    pre = _normalizar_turmas(pre)

    print("PÓS -> colunas:", pos.shape[1], "| respondentes:", pos.shape[0])
    print("PRÉ -> colunas:", pre.shape[1], "| respondentes:", pre.shape[0])

    return pos, pre


pos, pre = carregar_pipeline()

# ==============================================================================
# 2. METADADOS DAS PERGUNTAS (rótulos + enunciados completos)
# ==============================================================================
# Gerado pelo sigilo.py a partir dos headers originais do Forms (antes da
# anonimização). É a única fonte de verdade; não há mais extração manual nem
# tentativa de extrair dos CSVs anonimizados, já que dataset/*_anonimizado.csv
# só tem colunas curtas (Q01, Q02...), sem texto de header sobrando.
with open(os.path.join("dataset", "metadados_perguntas.json"), encoding="utf-8") as f:
    METADADOS_PERGUNTAS = json.load(f)

# Overrides manuais — só para itens que o Forms não etiqueta com rótulo entre
# colchetes (perguntas abertas de texto livre). Deixe vazio se não houver
# nenhuma exceção pendente.
DESCRICOES_OVERRIDE = {
    "Q14": "Maior contribuição da IA",
    "Q15": "Dificuldades e ajustes metodológicos",
}

DESCRICOES = {
    cod: DESCRICOES_OVERRIDE.get(cod) or info.get("rotulo") or cod
    for cod, info in METADADOS_PERGUNTAS.items()
}
ENUNCIADOS_COMPLETOS = {
    cod: info.get("enunciado")
    for cod, info in METADADOS_PERGUNTAS.items()
    if info.get("enunciado")
}

print(f"{len(DESCRICOES)} rótulos e {len(ENUNCIADOS_COMPLETOS)} enunciados carregados de metadados_perguntas.json.")
assert len(DESCRICOES) >= 25, f"Só {len(DESCRICOES)} rótulos disponíveis — confira dataset/metadados_perguntas.json."

# ==============================================================================
# 3. FUNÇÕES AUXILIARES DE TABELA
# ==============================================================================
def tabela_itens_md(codigos, coluna_texto="Enunciado aplicado"):
    """Monta um DataFrame Item/Texto para uma lista de códigos, usando os
    rótulos e enunciados já carregados de metadados_perguntas.json."""
    linhas = [
        {"Item": f"`{cod}`", coluna_texto: f"**{DESCRICOES.get(cod, cod)}:** {ENUNCIADOS_COMPLETOS.get(cod, '—')}"}
        for cod in codigos
    ]
    return pd.DataFrame(linhas)


def extrair_opcoes_q09(serie_q09):
    """Varre as respostas de Q09 (texto livre com múltiplas seleções, no formato
    '[Q09_XX - Rótulo] Descrição, [Q09_YY - Rótulo] Descrição, ...') e monta
    automaticamente código -> (rótulo curto, descrição completa), agregando o
    conjunto de opções que aparece em qualquer linha de resposta."""
    opcoes = {}
    for valor in serie_q09.dropna():
        partes = re.split(r",\s*(?=\[Q09_)", str(valor))
        for parte in partes:
            m = re.match(r"\[(Q09_\d{2})\s*-\s*([^\]]+)\]\s*(.*)", parte.strip())
            if m:
                cod, rotulo, desc = m.groups()
                if cod not in opcoes:
                    opcoes[cod] = (rotulo.strip(), desc.strip().rstrip(","))
    return opcoes

def fmt_num_br(x, casas=3):
    """Formata número no padrão brasileiro (vírgula decimal). Medianas de escala
    Likert (1-5) costumam ser inteiras, mas o formato se adapta caso não sejam."""
    if pd.isna(x):
        return "—"
    if casas == 0 or float(x).is_integer():
        return str(int(round(x)))
    return f"{x:.{casas}f}".replace(".", ",")


def fmt_p_br(p):
    if pd.isna(p):
        return "—"
    return "<0,001" if p < 0.001 else f"{p:.3f}".replace(".", ",")

OPCOES_Q09 = extrair_opcoes_q09(pos["Q09"])
print(f"{len(OPCOES_Q09)} opções de Q09 identificadas automaticamente (esperado: 10).")

display(pos.head())

PÓS -> colunas: 28 | respondentes: 80
PRÉ -> colunas: 20 | respondentes: 104
31 rótulos e 31 enunciados carregados de metadados_perguntas.json.
10 opções de Q09 identificadas automaticamente (esperado: 10).


,ID_Anonimo,Carimbo de data/hora,Turma,Q01,Q02,Q03,Q04,Q05,Q06,Q07,Q08_01,Q08_02,Q08_03,Q08_04,Q08_05,Q08_06,Q08_07,Q08_08,Q08_09,Q08_10,Q08_11,Q09,Q10,Q11,Q12,Q13,Q14,Q15
0,Aluno_NA_POS_01,09/08/2026 20:00:04,DA1,3,4,4,3,3,5,4,4,3,4,5,5,3,3,3,1,4,5,[Q09_03 - EPs em Aula] Resolução dos Exercício...,4,3,3,3,NaN,NaN
1,Aluno_POS_01,09/08/2026 20:06:08,DA2,4,3,4,4,5,5,4,5,5,5,4,5,5,5,3,5,5,5,[Q09_02 - Simuladores] Simuladores interativos...,4,4,4,4,NaN,NaN
2,Aluno_POS_02,09/08/2026 20:11:04,DA2,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,[Q09_02 - Simuladores] Simuladores interativos...,5,4,4,4,NaN,NaN
3,Aluno_NA_POS_02,09/08/2026 20:11:44,DA2,5,1,1,2,3,4,3,2,1,2,4,4,3,4,3,4,4,5,[Q09_03 - EPs em Aula] Resolução dos Exercício...,3,3,3,3,NaN,NaN
4,Aluno_POS_03,09/08/2026 20:13:25,DA1,5,5,5,3,4,3,5,4,2,5,5,4,4,4,3,5,5,4,[Q09_02 - Simuladores] Simuladores interativos...,3,5,2,1,NaN,NaN


In [24]:
#| quarto-raw: false
#| label: tbl-contagem-turmas
#| tbl-cap: "Quantidade de respondentes por turma nos questionários PRÉ e PÓS."
#| echo: false     # esconde o código, mas mantém o output (tabela/gráfico)

from IPython.display import Markdown

# 1. Contagem de respondentes por turma em cada momento (PRÉ / PÓS)
contagem_pre = pre["Turma"].value_counts()
contagem_pos = pos["Turma"].value_counts()

turmas = sorted(set(contagem_pre.index) | set(contagem_pos.index))

# 2. Monta as linhas da tabela, turma a turma
resultados_turmas = [
    {
        "Turma": turma,
        "PRÉ (n)": int(contagem_pre.get(turma, 0)),
        "PÓS (n)": int(contagem_pos.get(turma, 0)),
    }
    for turma in turmas
]

# 3. Linha de Total, em negrito (markdown), ao final
total_pre = sum(r["PRÉ (n)"] for r in resultados_turmas)
total_pos = sum(r["PÓS (n)"] for r in resultados_turmas)
resultados_turmas.append({
    "Turma": "**Total**",
    "PRÉ (n)": f"**{total_pre}**",
    "PÓS (n)": f"**{total_pos}**",
})

df_turmas = pd.DataFrame(resultados_turmas)
#print(f"Turmas identificadas: {', '.join(turmas)}\n")
#print(f"Total de respondentes — PRÉ: {total_pre} | PÓS: {total_pos}\n")
Markdown(df_turmas.to_markdown(index=False, colalign=("left", "center", "center")))

| Turma     |  PRÉ (n)  |  PÓS (n)  |
|:----------|:---------:|:---------:|
| DA1       |    39     |    39     |
| DA2       |    46     |    30     |
| MCC       |    19     |    11     |
| **Total** |  **104**  |  **80**   |

## Itens do questionário PÓS

O questionário PÓS foi estruturado em cinco blocos. O **Bloco 1** reúne a identificação opcional do estudante e a seleção de uma das três turmas. O **Bloco 2**, apresentado na @tbl-pos-bloco2, aborda a **autopercepção de aprendizado e desempenho**. O **Bloco 3**, apresentado na @tbl-pos-bloco3 (`Q08`) e na @tbl-pos-q09 (`Q09`), avalia os **recursos didáticos**. O **Bloco 4**, apresentado na @tbl-pos-bloco4, investiga a percepção sobre a **integração da Inteligência Artificial (IA)** ao processo de ensino e aprendizagem. Por fim, o **Bloco 5**, apresentado na @tbl-pos-bloco5, reúne duas questões abertas.

Os itens avaliativos dos Blocos 2, 3 e 4 utilizam uma **escala *Likert* de cinco pontos**: 1 = *Discordo totalmente*; 2 = *Discordo*; 3 = *Indiferente*; 4 = *Concordo*; e 5 = *Concordo totalmente*. A questão `Q09`, pertencente ao Bloco 3, adota formato distinto, solicitando a **seleção de exatamente quatro recursos** considerados mais eficazes para a aprendizagem pelo estudante.


### Itens do Bloco 2 — Autopercepção de Aprendizado e Desempenho

A @tbl-pos-bloco2 apresenta os sete itens do Bloco 2, que avaliam a **autopercepção dos estudantes sobre o aprendizado e o desempenho** ao final da disciplina. Os seis primeiros itens abrangem capacidade de programação, aplicação dos conhecimentos em outras disciplinas e na atuação profissional, fundamentos matemáticos, autonomia no uso de bibliotecas e percepção sobre o *feedback* por IA. Esses itens possuem correspondência com questões do questionário **PRÉ** e serão analisados de forma pareada na seção seguinte, permitindo avaliar a evolução das percepções individuais. O item `Q07` apresenta uma avaliação global da evolução dos conhecimentos teóricos e práticos em PDI-VC ao longo da disciplina.

In [25]:
#| label: tbl-pos-bloco2
#| tbl-cap: "Itens do Bloco 2 — Autopercepção de Aprendizado e Desempenho."
#| tbl-colwidths: "[8,80]"
#| echo: true

codigos_bloco2 = ["Q01", "Q02", "Q03", "Q04", "Q05", "Q06", "Q07"]
df_bloco2 = tabela_itens_md(codigos_bloco2)
Markdown(df_bloco2.to_markdown(index=False, colalign=("left", "left")))

| Item   | Enunciado aplicado                                                                                                                                                                                      |
|:-------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `Q01`  | **PÓS - Capacidade de Programação:** Atualmente, consigo resolver problemas computacionais básicos usando programação.                                                                                  |
| `Q02`  | **PÓS - Atuação Acadêmica:** Os conhecimentos desenvolvidos nesta disciplina têm me ajudado na minha atuação acadêmica em outras disciplinas.                                                           |
| `Q03`  | **PÓS - Atuação Profissional:** Os conhecimentos desenvolvidos nesta disciplina serão úteis para a minha atuação profissional                                                                           |
| `Q04`  | **PÓS - Fundamentos Matemáticos:** Atualmente, considero que minha base em Álgebra Linear e Cálculo (conceitos usados em PDI) é muito boa.                                                              |
| `Q05`  | **PÓS - Autonomia com Bibliotecas:** Atualmente, sinto que tenho boa autonomia para utilizar bibliotecas de processamento de dados/imagens (ex: OpenCV, NumPy, Matplotlib, morph.py).                   |
| `Q06`  | **PÓS - Feedback por IA:** Atualmente, considero útil o feedback automático de código gerado por IA.                                                                                                    |
| `Q07`  | **GLOBAL_BLOCO2 - Avaliação Geral do Aprendizado:** De forma geral, considero que o meu nível de conhecimento teórico e prático na área de PDI-VC evoluiu significativamente ao longo desta disciplina. |

### Itens do Bloco 3 — Recursos Didáticos

O **Bloco 3** avalia os recursos didáticos utilizados na disciplina. É composto por dez itens de avaliação (`Q08_01`--`Q08_10`), apresentados na @tbl-pos-bloco3; uma questão sobre o acesso ao VPL (`Q08_11`); uma questão de **preferência pelos recursos** (`Q09`); e uma **avaliação global** (`Q10`).


In [26]:
#| label: tbl-pos-bloco3
#| tbl-cap: "Itens do Bloco 3 — Recursos Didáticos."
#| tbl-colwidths: "[8,80]"
#| echo: false     # esconde o código, mas mantém o output (tabela/gráfico)

codigos_bloco3 = [f"Q08_{i:02d}" for i in range(1, 12)] + ["Q09", "Q10"]
df_bloco3 = tabela_itens_md(codigos_bloco3)
Markdown(df_bloco3.to_markdown(index=False, colalign=("left", "left")))

| Item     | Enunciado aplicado                                                                                                                                                                                                                                                                       |
|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `Q08_01` | **Livro Interativo:** O uso do Livro Interativo (HTML/IPYNB/PDF) com simuladores e teoria integrada contribuiu significativamente para o meu aprendizado.                                                                                                                                |
| `Q08_02` | **Material Didático por IA:** Os vídeos curtos (~7 min) e os slides (~12 slides) produzidos via Gemini Notebook ajudaram a introduzir e sintetizar o conteúdo antes do início da aula prática no Colab                                                                                   |
| `Q08_03` | **Simuladores Interativos:** A manipulação visual de parâmetros em tempo real nos simuladores interativos (no HTML/Colab) facilitou a compreensão dos algoritmos.                                                                                                                        |
| `Q08_04` | **Resolução de EPs em Aula:** A resolução acompanhada dos Exercícios de Programação (EPs) durante o horário de aula foi fundamental para o meu desempenho prático.                                                                                                                       |
| `Q08_05` | **Biblioteca morph.py:** O uso da biblioteca didática morph.py (com implementações didáticas como mm.dil0 e otimizadas) tornou o código dos algoritmos mais transparente e compreensível.                                                                                                |
| `Q08_06` | **Provas e Simulados Paramétricos:** As avaliações parametrizadas pelo MCTest e aplicadas via SEB (com questões individuais) foram eficazes para avaliar meu aprendizado de forma justa.                                                                                                 |
| `Q08_07` | **Feedback Socrático no VPL:** O feedback qualitativo/socrático gerado por IA ao pedir avaliação do código nos EPs do VPL/Moodle ajudou a identificar e corrigir erros na minha lógica.                                                                                                  |
| `Q08_08` | **Rubrica por IA em Simulados/Provas:** O relatório enviado por e-mail com a rubrica detalhada por IA e o resumo comparativo agregou valor ao meu processo de revisão das provas e simulados.                                                                                            |
| `Q08_09` | **Correção Objetiva VPL:** A validação automática objetiva por casos de teste no VPL/Moodle deu um retorno rápido e claro sobre o funcionamento do código.                                                                                                                               |
| `Q08_10` | **Uso do TestSuite.py:** O testsuite.py facilitou testar meus EPs no Colab antes de submeter no VPL.                                                                                                                                                                                     |
| `Q08_11` | **VPL só na UFABC:** O bloqueio de acesso ao VPL fora da rede da UFABC dificultou meus estudos.                                                                                                                                                                                          |
| `Q09`    | **Ranking de Preferência:** Escolha até 4 recursos que considerou MAIS EFICAZES no seu processo de estudo:                                                                                                                                                                               |
| `Q10`    | **GLOBAL_BLOCO3 - Avaliação Geral dos Recursos Didáticos:** Em conjunto, o ecossistema de recursos metodológicos oferecidos nesta disciplina (livro interativo, materiais multimídia por IA, simuladores, biblioteca didática e avaliações) foi altamente eficaz para o meu aprendizado. |

A questão `Q09` utilizou um formato de **seleção de preferência**, no qual cada estudante deveria selecionar **exatamente quatro recursos** considerados mais eficazes para seu processo de estudo. A questão apresentou **dez opções de recursos**, relacionadas na @tbl-pos-q09.


In [27]:
#| label: tbl-pos-q09
#| tbl-cap: "Opções apresentadas na questão `Q09` — *Ranking* agregado por frequência de seleção."
#| tbl-colwidths: "[8,80]"
#| echo: false     # esconde o código, mas mantém o output (tabela/gráfico)

linhas_q09 = [
    {"Item": f"`{cod}`", "Recurso": f"**{rotulo}:** {desc}"}
    for cod, (rotulo, desc) in sorted(OPCOES_Q09.items())
]
df_q09 = pd.DataFrame(linhas_q09)
Markdown(df_q09.to_markdown(index=False, colalign=("left", "left")))

| Item     | Recurso                                                                                                            |
|:---------|:-------------------------------------------------------------------------------------------------------------------|
| `Q09_01` | **Gemini Audio/Visual:** Material didático audiovisual gerado por IA (Vídeos e Slides do Gemini Notebook)          |
| `Q09_02` | **Simuladores:** Simuladores interativos (manipulação de parâmetros)                                               |
| `Q09_03` | **EPs em Aula:** Resolução dos Exercícios de Programação (EPs) em sala de aula                                     |
| `Q09_04` | **Livro Interativo:** Livro interativo no formato HTML / Colab                                                     |
| `Q09_05` | **morph.py:** Biblioteca didática morph.py                                                                         |
| `Q09_06` | **IA Feedback VPL:** Feedback socrático por IA a cada submissão de código no VPL/Moodle                            |
| `Q09_07` | **IA Rubrica Provas:** Feedback por e-mail com rubrica de avaliação qualitativa gerada por IA nos Simulados/Provas |
| `Q09_08` | **SEB Simulados:** Simulados quinzenais em laboratório (SEB)                                                       |
| `Q09_09` | **VPL Casos Teste:** Validação objetiva por casos de teste no VPL                                                  |
| `Q09_10` | **TestSuite.py:** Uso do testsuite.py no Colab para testar EPs localmente                                          |

### Itens do Bloco 4 — Avaliação Específica do Uso Transversal da Inteligência Artificial

Os itens `Q11`--`Q13`, apresentados na @tbl-pos-bloco4, avaliam a percepção dos estudantes sobre a **integração transversal da Inteligência Artificial (IA)** no processo de ensino e aprendizagem. Como utilizam a mesma escala *Likert* de cinco pontos descrita anteriormente, esses itens compõem, juntamente com os demais itens ordinais, a síntese dos **21 itens *Likert*** apresentada ao final deste apêndice.

O item `Q08_11` e a questão `Q09`, pertencentes ao **Bloco 3**, não integram essa síntese, pois apresentam formatos e finalidades distintos da avaliação ordinal dos recursos didáticos. O `Q08_11` investiga uma dificuldade específica relacionada ao acesso ao VPL, enquanto `Q09` utiliza **seleção de exatamente quatro recursos**, sem estabelecer ordenação entre as alternativas.


In [28]:
#| label: tbl-pos-bloco4
#| tbl-cap: "Itens do Bloco 4 — Avaliação Específica do Uso Transversal da Inteligência Artificial"
#| tbl-colwidths: "[8,80]"
#| echo: false     # esconde o código, mas mantém o output (tabela/gráfico)

codigos_bloco4 = ["Q11", "Q12", "Q13"]
df_bloco4 = tabela_itens_md(codigos_bloco4)
Markdown(df_bloco4.to_markdown(index=False, colalign=("left", "left")))

| Item   | Enunciado aplicado                                                                                                                                                                                          |
|:-------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `Q11`  | **Pensamento Crítico e Alucinações:** Perceber eventuais limitações ou imprecisões nas explicações da IA (no feedback ou materiais) estimulou meu senso crítico sobre a correção do código.                 |
| `Q12`  | **Transversalidade Pedagógica:** Recomendo que essa abordagem integrada de IA (produção de material + feedback socrático no VPL + rubrica qualitativa em provas) seja adotada em outras disciplinas.        |
| `Q13`  | **GLOBAL_BLOCO4 - Avaliação Geral da Integração de IA:** O uso transversal e transparente de Inteligência Artificial ao longo da disciplina agregou valor significativo à minha experiência de aprendizado. |

### Bloco 5 — Questões Abertas

O **Bloco 5** reúne duas questões abertas e opcionais, apresentadas na @tbl-pos-bloco5, destinadas a complementar a avaliação quantitativa com *feedback* qualitativo sobre a experiência dos estudantes com a metodologia da disciplina e, especificamente, sobre o uso da **Inteligência Artificial (IA)** no processo de ensino e aprendizagem.


In [29]:
#| label: tbl-pos-bloco5
#| tbl-cap: "Itens do Bloco 5 — Questões Abertas."
#| tbl-colwidths: "[8,80]"
#| echo: false     # esconde o código, mas mantém o output (tabela/gráfico)

codigos_bloco5 = ["Q14", "Q15"]
df_bloco5 = tabela_itens_md(codigos_bloco5)
Markdown(df_bloco5.to_markdown(index=False, colalign=("left", "left")))

| Item   | Enunciado aplicado                                                                                                                                                                                              |
|:-------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `Q14`  | **Maior contribuição da IA:** Dentre os usos da IA nesta disciplina (geração de material, feedback socrático nos EPs, rubrica nos simulados/provas), qual trouxe o maior ganho para o seu aprendizado? Por quê? |
| `Q15`  | **Dificuldades e ajustes metodológicos:** Qual elemento da metodologia (SEB, MCTest, tempo de prova, feedback por IA, biblioteca) apresentou maior dificuldade ou precisará de ajustes nas próximas ofertas?    |

As questões `Q14` e `Q15` não utilizaram escala ordinal nem alternativas predefinidas. As respostas, de caráter **aberto e opcional**, permitiram aos estudantes registrar livremente suas percepções, justificativas, dificuldades e sugestões de melhoria. Por esse motivo, esses itens não integram as análises estatísticas baseadas nas questões *Likert* e são examinados separadamente por meio de **codificação temática** e **classificação de valência**, apresentadas nas seções seguintes.


## Perfil acadêmico dos participantes

A interpretação dos resultados deve considerar o **perfil acadêmico dos participantes**, especialmente a composição da amostra de graduação, formada majoritariamente por estudantes em **etapas avançadas da formação**. No questionário PRÉ, **88 estudantes indicaram Ciência da Computação** e **30, Ciência de Dados** como cursos que estavam cursando ou pretendiam cursar após o Bacharelado Interdisciplinar. Como era permitida a seleção de mais de uma opção, esses quantitativos não são mutuamente exclusivos, conforme apresentado na @tbl-formacao-graduacao.

Essa composição ajuda a contextualizar a elevada autopercepção de **capacidade de programação** (`Q01`) observada no PRÉ e no PÓS, analisada nas subseções seguintes. Por outro lado, **Processamento Digital de Imagens (PDI) não constitui componente obrigatório da matriz curricular de todos esses cursos**; portanto, a formação acadêmica não permite presumir experiência prévia específica nessa área.

Na **pós-graduação**, observa-se maior diversidade nas formações de origem, incluindo Física, Engenharia Biomédica, Matemática, Ciências Atuariais, Biotecnologia, Computação e Ciência de Dados. Essa heterogeneidade corresponde a trajetórias acadêmicas distintas e pode estar associada a diferentes níveis de experiência prévia em programação e processamento de imagens.

Assim, a elevada autopercepção de **capacidade de programação** no PRÉ (ver @tbl-pre-pos-wilcoxon) deve ser interpretada considerando a composição da amostra. A concentração de estudantes de graduação em etapas avançadas da formação pode ter contribuído para um possível **efeito teto** em `Q01`, cuja percepção já era elevada no início da disciplina e permaneceu em nível alto nos dois momentos.


In [31]:
#| label: tbl-formacao-graduacao
#| tbl-cap: "Cursos indicados pelos estudantes de graduação. Percentuais calculados sobre os 104 respondentes. Como a questão permitia múltiplas respostas, os percentuais não são mutuamente exclusivos e podem somar mais de 100%."
#| echo: false

def contar_selecoes_multiplas(serie, separador=","):
    """Conta a frequência de cada opção numa coluna de múltipla seleção do Forms,
    no formato 'Opção A, Opção B, Opção C'. Cada resposta pode conter 0+ opções."""
    contagem = {}
    for valor in serie.dropna().astype(str):
        for item in valor.split(separador):
            item = item.strip()
            if item:
                contagem[item] = contagem.get(item, 0) + 1
    return contagem


col_cursos_candidatos = [c for c in pre.columns if "curso" in c.lower() and "bacharelado" in c.lower()]
assert col_cursos_candidatos, "Coluna de cursos não encontrada em pre.columns — confira o nome exato do header no CSV."
col_cursos = col_cursos_candidatos[0]

contagem_cursos = contar_selecoes_multiplas(pre[col_cursos])
n_total_pre = len(pre)  # 104 — denominador fixo, conforme nota da tabela

cursos_frequentes = {c: n for c, n in contagem_cursos.items() if n > 1}
cursos_unicos = sorted(c for c, n in contagem_cursos.items() if n == 1)

resultados_cursos = [
    {
        "Curso/área indicada": curso,
        "Estudantes (n)": n,
        "Percentual*": f"{n / n_total_pre:.1%}".replace(".", ","),
    }
    for curso, n in sorted(cursos_frequentes.items(), key=lambda x: -x[1])
]

if cursos_unicos:
    resultados_cursos.append({
        "Curso/área indicada": "Outros (1 estudante cada): " + ", ".join(cursos_unicos),
        "Estudantes (n)": len(cursos_unicos),
        "Percentual*": f"{1 / n_total_pre:.1%}".replace(".", ",") + " cada",
    })

df_cursos = pd.DataFrame(resultados_cursos)
Markdown(df_cursos.to_markdown(index=False, colalign=("left", "center", "center")))

| Curso/área indicada                                                                                                                                                                                        |  Estudantes (n)  |  Percentual*  |
|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------:|:-------------:|
| Ciência da Computação                                                                                                                                                                                      |        88        |     84,6%     |
| Ciência de Dados                                                                                                                                                                                           |        30        |     28,8%     |
| Instrumentação                                                                                                                                                                                             |        8         |     7,7%      |
| Automação e Robótica                                                                                                                                                                                       |        8         |     7,7%      |
| Matemática                                                                                                                                                                                                 |        6         |     5,8%      |
| Informação                                                                                                                                                                                                 |        6         |     5,8%      |
| Biomédica                                                                                                                                                                                                  |        5         |     4,8%      |
| Física                                                                                                                                                                                                     |        4         |     3,8%      |
| Neurociência                                                                                                                                                                                               |        3         |     2,9%      |
| Gestão                                                                                                                                                                                                     |        2         |     1,9%      |
| Outros (1 estudante cada): Aeroespacial, Ambiental e Urbana, Biotecnologia, Ciências Atuariais, Ciências Biológicas, Ciências atuariais, Energia, Filosofia, Pós em Ciência da Computação, formado em 1997 |        10        |   1,0% cada   |

## Comparação entre os Questionários PRÉ e PÓS {#sec-pre-pos}

A comparação entre os questionários **PRÉ** e **PÓS** busca identificar mudanças nas percepções dos estudantes ao longo da disciplina. A análise contempla seis dimensões: capacidade de programação, *feedback* por IA, atuação acadêmica, atuação profissional, fundamentos matemáticos e autonomia no uso de bibliotecas.

Para avaliar essas dimensões, foram empregados dois procedimentos complementares. O teste de *Wilcoxon* foi aplicado aos **15 estudantes pareados**, permitindo avaliar mudanças nas percepções individuais. O teste de *Mann--Whitney U* foi aplicado às **amostras completas**, permitindo verificar diferenças na distribuição das respostas entre os dois momentos. Dessa forma, o primeiro procedimento avalia **mudanças intraindividuais**, enquanto o segundo identifica **diferenças entre as amostras**, cuja composição não é necessariamente a mesma.


### Evolução da Percepção dos Estudantes: PRÉ versus PÓS

A comparação entre os questionários **PRÉ** e **PÓS** busca identificar mudanças nas percepções dos estudantes ao longo da disciplina. Foram obtidas **104 respostas no PRÉ** e **80 no PÓS**, com **15 estudantes identificados nos dois momentos**, constituindo a amostra pareada utilizada na análise da evolução individual.

A análise contempla seis dimensões:

* **Capacidade de programação** (`PRE_Q01 → Q01`);
* ***Feedback* por IA** (`PRE_Q02 → Q06`);
* **Atuação acadêmica** (`PRE_Q03 → Q02`);
* **Atuação profissional** (`PRE_Q04 → Q03`);
* **Fundamentos matemáticos** (`PRE_Q05 → Q04`);
* **Autonomia com bibliotecas** (`PRE_Q06 → Q05`).

As médias e medianas apresentadas na @tbl-pre-pos-wilcoxon referem-se às **amostras completas** (PRÉ $N=104$; PÓS $N=80$) e têm finalidade exclusivamente descritiva, correspondendo aos mesmos valores apresentados na @tbl-pre-pos-mw. A evolução individual foi avaliada pelo teste de *Wilcoxon*, aplicado aos **15 estudantes pareados**. Como esse teste é baseado nas diferenças ordenadas entre as observações, a **mediana** é apresentada juntamente com a média, permitindo uma interpretação mais adequada da tendência central em conjunto com o resultado inferencial.


In [32]:
#| label: tbl-pre-pos-wilcoxon
#| tbl-cap: "Comparação descritiva (média e mediana) das amostras completas e teste de Wilcoxon para a amostra pareada de 15 estudantes."
#| echo: false

MAPEAMENTO_PRE_POS = [
    ("Capacidade de programação", "PRE_Q01", "Q01"),
    ("Feedback por IA",           "PRE_Q02", "Q06"),
    ("Atuação acadêmica",         "PRE_Q03", "Q02"),
    ("Atuação profissional",      "PRE_Q04", "Q03"),
    ("Fundamentos matemáticos",   "PRE_Q05", "Q04"),
    ("Autonomia com bibliotecas", "PRE_Q06", "Q05"),
]


def fmt_num_br(x, casas=3):
    """Formata número no padrão brasileiro (vírgula decimal). Medianas de escala
    Likert (1-5) costumam ser inteiras, mas o formato se adapta caso não sejam."""
    if pd.isna(x):
        return "—"
    if casas == 0 or float(x).is_integer():
        return str(int(round(x)))
    return f"{x:.{casas}f}".replace(".", ",")


def fmt_p_br(p):
    if pd.isna(p):
        return "—"
    return "<0,001" if p < 0.001 else f"{p:.3f}".replace(".", ",")


def estatisticas_descritivas_pre_pos(col_pre, col_pos):
    """Média e mediana de uma dimensão, sobre as amostras completas de PRÉ e PÓS
    (não só os pareados). Reaproveitada também na tbl-pre-pos-mw, que reporta os
    mesmos valores descritivos."""
    serie_pre = pd.to_numeric(pre[col_pre], errors="coerce").dropna()
    serie_pos = pd.to_numeric(pos[col_pos], errors="coerce").dropna()
    return {
        "media_pre": serie_pre.mean(), "mediana_pre": serie_pre.median(),
        "media_pos": serie_pos.mean(), "mediana_pos": serie_pos.median(),
    }


# Amostra pareada: junção por ID_Anonimo. Só sobrevivem no merge os estudantes
# com o mesmo ID nos dois dataframes -- ou seja, os identificados como
# Aluno_PAR_XX durante a anonimização (sigilo.py).
pareado = pre.merge(pos, on="ID_Anonimo", suffixes=("_pre", "_pos"))
n_pareados = len(pareado)
n_pre_total = len(pre)
n_pos_total = len(pos)

assert n_pareados == 15, f"Esperava 15 estudantes pareados, encontrei {n_pareados} — confira a auditoria de pareamento."

resultados_evolucao = []
for label, col_pre, col_pos in MAPEAMENTO_PRE_POS:
    desc = estatisticas_descritivas_pre_pos(col_pre, col_pos)

    par_pre = pd.to_numeric(pareado[col_pre], errors="coerce")
    par_pos = pd.to_numeric(pareado[col_pos], errors="coerce")
    validos = par_pre.notna() & par_pos.notna()
    _, p_valor = stats.wilcoxon(par_pre[validos], par_pos[validos])

    resultados_evolucao.append({
        "Dimensão": label,
        f"Média PRÉ (N={n_pre_total})": fmt_num_br(desc["media_pre"]),
        f"Média PÓS (N={n_pos_total})": fmt_num_br(desc["media_pos"]),
        "Mediana PRÉ": fmt_num_br(desc["mediana_pre"], casas=0),
        "Mediana PÓS": fmt_num_br(desc["mediana_pos"], casas=0),
        f"p Wilcoxon (pareado, n={n_pareados})": fmt_p_br(p_valor),
        "Interpretação": "Evidência de mudança" if p_valor < 0.05 else "Sem evidência de mudança",
    })

df_evolucao = pd.DataFrame(resultados_evolucao)
Markdown(df_evolucao.to_markdown(index=False, colalign=("left",) + ("center",) * 6))

| Dimensão                  |  Média PRÉ (N=104)  |  Média PÓS (N=80)  |  Mediana PRÉ  |  Mediana PÓS  |  p Wilcoxon (pareado, n=15)  |      Interpretação       |
|:--------------------------|:-------------------:|:------------------:|:-------------:|:-------------:|:----------------------------:|:------------------------:|
| Capacidade de programação |        4,481        |       4,562        |       5       |       5       |            0,655             | Sem evidência de mudança |
| Feedback por IA           |        3,981        |       3,438        |       4       |       4       |            0,070             | Sem evidência de mudança |
| Atuação acadêmica         |        4,385        |       3,712        |       5       |       4       |            0,083             | Sem evidência de mudança |
| Atuação profissional      |        4,356        |       3,725        |       5       |       4       |            0,052             | Sem evidência de mudança |
| Fundamentos matemáticos   |        3,212        |       3,538        |       3       |       4       |            0,272             | Sem evidência de mudança |
| Autonomia com bibliotecas |        3,663        |       3,825        |       4       |       4       |            0,885             | Sem evidência de mudança |

Os resultados do teste de *Wilcoxon* **não forneceram evidência estatisticamente significativa de mudança sistemática** entre PRÉ e PÓS em nenhuma das seis dimensões ($p>0,05$). Essa ausência de significância não demonstra que não tenha ocorrido mudança; indica apenas que, na **amostra pareada de 15 estudantes**, não foi obtida evidência estatística suficiente para detectá-la.

A **capacidade de programação** apresentou médias elevadas nos dois momentos, passando de **4,481 para 4,562**, com mediana igual a 5 em ambos. O teste pareado ($p=0,655$) não indicou evidência de mudança sistemática, resultado compatível com a elevada percepção inicial observada para essa dimensão.

O ***feedback* por IA** apresentou redução descritiva da média, de **3,981 para 3,438**, embora a diferença não tenha sido estatisticamente significativa ($p=0,070$). Também foram observadas reduções em **atuação acadêmica**, de 4,385 para 3,712 ($p=0,083$), e **atuação profissional**, de 4,356 para 3,725 ($p=0,052$). Apesar das variações descritivas, os resultados não permitem afirmar que tenha ocorrido mudança sistemática nessas dimensões.

Em sentido oposto, **fundamentos matemáticos** apresentou aumento da média, de **3,212 para 3,538** ($p=0,272$), enquanto a **autonomia com bibliotecas** passou de **3,663 para 3,825** ($p=0,885$). Novamente, as diferenças observadas não alcançaram significância estatística.

As reduções descritivas em **atuação acadêmica** e **atuação profissional** podem ser compatíveis com uma possível **recalibração das expectativas e da autoavaliação**, à medida que os estudantes entram em contato com problemas concretos de PDI-VC. Essa interpretação deve, contudo, ser considerada apenas uma **hipótese pedagógica**, e não uma conclusão causal derivada dos resultados estatísticos.


### Análise complementar entre as amostras completas

Como análise complementar, o teste de *Mann--Whitney U* foi aplicado às **amostras completas** do PRÉ ($N=104$) e do PÓS ($N=80$), sem considerar o pareamento entre indivíduos. Os resultados são apresentados na @tbl-pre-pos-mw e permitem verificar se as distribuições das respostas diferem entre os dois momentos, independentemente da correspondência individual entre os participantes.


In [33]:
#| label: tbl-pre-pos-mw
#| tbl-cap: "Comparação entre as amostras completas PRÉ (N=104) e PÓS (N=80) pelo teste de Mann-Whitney U. O d de Cohen é apresentado como medida padronizada da diferença entre as médias, enquanto a significância estatística foi avaliada pelo teste de Mann-Whitney U."
#| echo: false

def cohen_d(x, y):
    """d de Cohen para duas amostras independentes, usando o desvio-padrão
    combinado (pooled). Sinal positivo indica PÓS > PRÉ."""
    nx, ny = len(x), len(y)
    var_pooled = ((nx - 1) * x.var(ddof=1) + (ny - 1) * y.var(ddof=1)) / (nx + ny - 2)
    return (y.mean() - x.mean()) / np.sqrt(var_pooled)


def classificar_efeito(d):
    ad = abs(d)
    if ad < 0.20:
        return "Muito pequeno"
    if ad < 0.50:
        return "Pequeno"
    if ad < 0.80:
        return "Médio"
    return "Grande"


resultados_mw = []
for label, col_pre, col_pos in MAPEAMENTO_PRE_POS:
    desc = estatisticas_descritivas_pre_pos(col_pre, col_pos)
    serie_pre = pd.to_numeric(pre[col_pre], errors="coerce").dropna()
    serie_pos = pd.to_numeric(pos[col_pos], errors="coerce").dropna()

    d = cohen_d(serie_pre, serie_pos)
    _, p_valor = stats.mannwhitneyu(serie_pre, serie_pos, alternative="two-sided")

    resultados_mw.append({
        "Dimensão": label,
        "Média PRÉ": fmt_num_br(desc["media_pre"]),
        "Média PÓS": fmt_num_br(desc["media_pos"]),
        "Mediana PRÉ": fmt_num_br(desc["mediana_pre"], casas=0),
        "Mediana PÓS": fmt_num_br(desc["mediana_pos"], casas=0),
        "d de Cohen": fmt_num_br(d),
        "p-valor": fmt_p_br(p_valor),
        "Efeito": classificar_efeito(d),
        "_p_num": p_valor,  # coluna auxiliar só para ordenar; removida antes de exibir
    })

df_mw = pd.DataFrame(resultados_mw).sort_values("_p_num").drop(columns="_p_num")
Markdown(df_mw.to_markdown(index=False, colalign=("left",) + ("center",) * 7))

| Dimensão                  |  Média PRÉ  |  Média PÓS  |  Mediana PRÉ  |  Mediana PÓS  |  d de Cohen  |  p-valor  |    Efeito     |
|:--------------------------|:-----------:|:-----------:|:-------------:|:-------------:|:------------:|:---------:|:-------------:|
| Atuação acadêmica         |    4,385    |    3,712    |       5       |       4       |    -0,775    |  <0,001   |     Médio     |
| Atuação profissional      |    4,356    |    3,725    |       5       |       4       |    -0,677    |  <0,001   |     Médio     |
| Feedback por IA           |    3,981    |    3,438    |       4       |       4       |    -0,471    |   0,008   |    Pequeno    |
| Fundamentos matemáticos   |    3,212    |    3,538    |       3       |       4       |    0,303     |   0,048   |    Pequeno    |
| Capacidade de programação |    4,481    |    4,562    |       5       |       5       |    0,108     |   0,301   | Muito pequeno |
| Autonomia com bibliotecas |    3,663    |    3,825    |       4       |       4       |    0,142     |   0,865   | Muito pequeno |

Para a interpretação do $d$ de *Cohen*, foram adotados, de forma aproximada, os seguintes intervalos: valores absolutos inferiores a 0,20 foram classificados como **muito pequenos**; entre 0,20 e 0,49, como **pequenos**; entre 0,50 e 0,79, como **médios**; e iguais ou superiores a 0,80, como **grandes**.

A comparação entre as amostras completas identificou diferenças estatisticamente significativas em **atuação acadêmica** ($p<0,001$), **atuação profissional** ($p<0,001$), ***feedback* por IA** ($p=0,008$) e **fundamentos matemáticos** ($p=0,048$). Os maiores tamanhos de efeito foram observados em atuação acadêmica ($d=-0,775$) e atuação profissional ($d=-0,677$), ambos classificados como **médios**. Em *feedback* por IA, o efeito foi **pequeno** ($d=-0,471$), assim como em fundamentos matemáticos ($d=0,303$).

Em contraste, **capacidade de programação** ($d=0,108$; $p=0,301$) e **autonomia com bibliotecas** ($d=0,142$; $p=0,865$) não apresentaram diferenças estatisticamente significativas, com efeitos **muito pequenos**.

Esses resultados **não devem ser interpretados como evidência de evolução individual**, pois o teste de *Mann--Whitney U* compara amostras independentes. As diferenças observadas podem decorrer tanto de mudanças entre os momentos quanto da composição distinta dos respondentes. Assim, os resultados caracterizam **diferenças entre os conjuntos de participantes do PRÉ e do PÓS**, e não efeitos atribuíveis à disciplina. Para avaliar mudanças intraindividuais, devem ser considerados os resultados do teste de *Wilcoxon* aplicado aos **15 estudantes pareados**.


## Avaliação dos Recursos Didáticos no PÓS — Teste de Friedman 

O Bloco 3 do questionário PÓS avalia **dez recursos didáticos** utilizados durante a disciplina, correspondentes aos itens `Q08_01`–`Q08_10`. Os recursos representam práticas pedagógicas distintas: alguns não utilizaram IA diretamente, outros utilizaram IA como ferramenta de apoio e três utilizaram IA diretamente na geração de conteúdo, *feedback* ou rubricas.
Foram considerados **$N=80$ estudantes**, cada um avaliando os dez recursos.

O teste de Friedman [@friedman1937] indicou diferença estatisticamente significativa entre as avaliações dos dez recursos, $χ²(9) = 189,299$, $p < 0,001$, rejeitando a hipótese de que todos apresentassem avaliações equivalentes. O resultado demonstra, portanto, que os estudantes **diferenciaram os recursos quanto à sua avaliação**.

O **W de Kendall** [@kendall1945] apresentou valor $W = 0,266$, indicando uma concordância limitada entre os estudantes quanto à ordenação dos recursos. Assim, embora exista diferença estatisticamente significativa entre as avaliações, não há forte consenso entre os estudantes sobre a ordem de preferência dos dez recursos.

## Avaliação dos Recursos Didáticos no PÓS — Teste de Friedman 

O Bloco 3 do questionário PÓS avalia **dez recursos didáticos** utilizados durante a disciplina, correspondentes aos itens `Q08_01`–`Q08_10`. Os recursos representam práticas pedagógicas distintas: alguns não utilizaram IA diretamente, outros utilizaram IA como ferramenta de apoio e três utilizaram IA diretamente na geração de conteúdo, *feedback* ou rubricas.
Foram considerados **N=`{python} n_friedman` estudantes**, cada um avaliando os dez recursos, apresentados na @tbl-bloco3-friedman.

O teste de Friedman [@friedman1937] indicou diferença estatisticamente significativa entre as avaliações dos dez recursos, χ²(`{python} df_friedman`) = `{python} f"{chi2_friedman:.3f}".replace(".", ",")`, `{python} fmt_p_br(p_friedman)`, rejeitando a hipótese de que todos apresentassem avaliações equivalentes. O resultado demonstra, portanto, que os estudantes **diferenciaram os recursos quanto à sua avaliação**.

O **W de Kendall** [@kendall1945] apresentou valor W = `{python} f"{w_kendall:.3f}".replace(".", ",")`, indicando uma concordância limitada entre os estudantes quanto à ordenação dos recursos. Assim, embora exista diferença estatisticamente significativa entre as avaliações, não há forte consenso entre os estudantes sobre a ordem de preferência dos dez recursos.


In [34]:
#| label: tbl-bloco3-friedman
#| tbl-cap: "Ranking médio dos dez recursos didáticos avaliados no Bloco 3 (Q08_01–Q08_10), ordenado da avaliação mais alta para a mais baixa."
#| echo: false

codigos_bloco3_friedman = [f"Q08_{i:02d}" for i in range(1, 11)]

# Casos completos: só entram estudantes que responderam aos 10 itens
dados_friedman = pos[codigos_bloco3_friedman].apply(pd.to_numeric, errors="coerce").dropna()
n_friedman = len(dados_friedman)

assert n_friedman == 80, f"Esperava N=80 no teste de Friedman, encontrei {n_friedman} — confira o filtro de qualidade do PÓS."

# Teste de Friedman (diferença global entre os 10 recursos)
chi2_friedman, p_friedman = stats.friedmanchisquare(*[dados_friedman[c] for c in codigos_bloco3_friedman])
k_friedman = len(codigos_bloco3_friedman)
df_friedman = k_friedman - 1

# W de Kendall, derivado da estatística de Friedman: W = chi2 / (n * (k-1))
w_kendall = chi2_friedman / (n_friedman * df_friedman)

print(f"Friedman: N={n_friedman}, chi2({df_friedman})={chi2_friedman:.3f}, p={p_friedman:.3g}")
print(f"W de Kendall: {w_kendall:.3f}")

# Ranking médio por item (mesma base de cálculo do teste: rank de 1 a 10
# atribuído por estudante, em cada linha, com empates resolvidos pela média)
ranks = dados_friedman.rank(axis=1, method="average", ascending=False)  # 1 = mais bem avaliado
rank_medio = ranks.mean()

resultados_friedman = [
    {
        "Item": f"`{cod}`",
        "Recurso": DESCRICOES.get(cod, cod),
        "Média": fmt_num_br(dados_friedman[cod].mean()),
        "Mediana": fmt_num_br(dados_friedman[cod].median(), casas=0),
        "Rank médio": fmt_num_br(rank_medio[cod]),
    }
    for cod in rank_medio.sort_values().index
]
df_friedman_tbl = pd.DataFrame(resultados_friedman)
Markdown(df_friedman_tbl.to_markdown(index=False, colalign=("left", "left", "center", "center", "center")))

Friedman: N=80, chi2(9)=189.662, p=4.85e-36
W de Kendall: 0.263


| Item     | Recurso                            |  Média  |  Mediana  |  Rank médio  |
|:---------|:-----------------------------------|:-------:|:---------:|:------------:|
| `Q08_04` | Resolução de EPs em Aula           |  4,588  |     5     |    4,088     |
| `Q08_05` | Biblioteca morph.py                |  4,300  |     5     |    4,550     |
| `Q08_03` | Simuladores Interativos            |  4,362  |     5     |    4,600     |
| `Q08_09` | Correção Objetiva VPL              |  4,213  |     5     |    4,888     |
| `Q08_01` | Livro Interativo                   |  4,287  |     4     |    4,900     |
| `Q08_10` | Uso do TestSuite.py                |  4,263  |     5     |    4,906     |
| `Q08_06` | Provas e Simulados Paramétricos    |  4,075  |     4     |    5,438     |
| `Q08_08` | Rubrica por IA em Simulados/Provas |  3,587  |     4     |    6,719     |
| `Q08_07` | Feedback Socrático no VPL          |  3,388  |     4     |    7,162     |
| `Q08_02` | Material Didático por IA           |  3,087  |     3     |    7,750     |

### Classificação dos recursos segundo o papel da IA 

Para interpretar adequadamente os resultados, os dez recursos avaliados em
`Q08` podem ser classificados em três grupos, conforme apresentado na
@tbl-classificacao-ia.

| Categoria                       | Recursos                                                                                                                                                       | Participação da IA                                |
| :------------------------------ | :------------------------------------------------------------------------------------------------------------------------------------------------------------- | :------------------------------------------------ |
| **Sem uso direto de IA**        | `Q08_04` — Resolução de EPs em Aula; `Q08_09` — Correção Objetiva no VPL                                                                                       | A IA não participa diretamente do recurso         |
| **IA como ferramenta de apoio** | `Q08_01` — Livro Interativo; `Q08_03` — Simuladores Interativos; `Q08_05` — Biblioteca `morph.py`; `Q08_06` — Provas e Simulados Paramétricos; `Q08_10` — `TestSuite.py` | Apoio à programação, preparação ou formatação     |
| **IA com participação direta**  | `Q08_02` — Material Didático por IA; `Q08_07` — *Feedback* Socrático no VPL; `Q08_08` — Rubrica por IA em Simulados/Provas                                           | Geração direta de conteúdo, *feedback* ou rubrica |

: Classificação dos dez recursos didáticos segundo o papel da Inteligência Artificial. {#tbl-classificacao-ia}

Essa classificação considera o papel da IA na produção, preparação ou
funcionamento dos recursos, e não apenas a interação direta do estudante com
a tecnologia. Assim, o emprego da IA como ferramenta de apoio à produção ou à
implementação de um recurso difere de sua participação direta na geração de
material didático, *feedback* ou rubricas.

### *Ranking* agregado por frequência de seleção dos recursos 

A ordenação das médias de `Q08`, acompanhada da mediana e do teste de diferença em relação ao valor neutro 3,0, está apresentada na @tbl-q08-ranking.

| Posição | Item     | Recurso                            | Média |   DP | Mediana | $p$ vs. 3,0 |
| :-----: | :------- | :--------------------------------- | ----: | ---: | ------: | ----------: |
|    1    | `Q08_04` | Resolução de EPs em aula           |  4,59 | 0,65 |       5 |    $<0,001$ |
|    2    | `Q08_03` | Simuladores interativos            |  4,35 | 0,85 |       5 |    $<0,001$ |
|    3    | `Q08_01` | Livro interativo                   |  4,30 | 0,81 |       4 |    $<0,001$ |
|    4    | `Q08_05` | Biblioteca `morph.py`              |  4,29 | 1,05 |       5 |    $<0,001$ |
|    5    | `Q08_10` | Uso do `TestSuite.py`              |  4,25 | 0,94 |       5 |    $<0,001$ |
|    6    | `Q08_09` | Correção objetiva no VPL           |  4,20 | 1,02 |       5 |    $<0,001$ |
|    7    | `Q08_06` | Provas e simulados paramétricos    |  4,09 | 1,08 |       4 |    $<0,001$ |
|    8    | `Q08_08` | Rubrica por IA em simulados/provas |  3,57 | 1,09 |       3 |    $<0,001$ |
|    9    | `Q08_07` | *Feedback* socrático no VPL        |  3,42 | 1,23 |       4 |       0,006 |
|    10   | `Q08_02` | Material didático por IA           |  3,08 | 1,37 |       3 |       0,758 |

: Avaliação dos dez recursos didáticos no questionário PÓS ($N=79$), ordenada pela média. O $p$-valor refere-se ao teste de postos sinalizados de *Wilcoxon* para uma amostra, tendo 3,0 como valor de referência. {#tbl-q08-ranking}

Os sete primeiros recursos apresentaram médias superiores a 4,0. A **Resolução de EPs em aula** obteve a maior avaliação (4,59), seguida pelos **Simuladores interativos** (4,35) e pelo **Livro interativo** (4,30).

Além disso, os sete primeiros recursos apresentaram médias significativamente superiores ao valor neutro de 3,0 (p < 0,001). A rubrica por IA (M = 3,57; p < 0,001) e o *feedback* socrático no VPL (M = 3,42; p = 0,006) também apresentaram avaliações superiores ao neutro, embora com médias menores. Já o material didático por IA (M = 3,08; p = 0,758) não diferiu estatisticamente do valor neutro, indicando uma avaliação essencialmente neutra.


Os três recursos com participação direta da IA na produção de conteúdo, *feedback* ou rubricas apresentaram as três menores médias entre os dez recursos avaliados:

* **Material didático por IA ([Apêndice D](#apendice-gemini-notebook)):** média 3,08;
* ***Feedback* socrático no VPL [@zampirolli2026intelligent]:** média 3,42;
* **Rubrica por IA em simulados/provas ([Apêndice E](#apendice-e-ia-avaliacao)):** média 3,57.

No caso do *feedback* socrático, é importante considerar que a experiência descrita em [@zampirolli2026intelligent] ocorreu em um contexto pedagógico distinto: uma disciplina introdutória de **Lógica de Programação**, com problemas de menor complexidade em comparação com os problemas de PDI-VC. Naquela aplicação, o *feedback* por IA era produzido a partir **exclusivamente do código submetido pelo estudante**, sem comentários ou a descrição do enunciado da questão, seguido de um *prompt* socrático. Em PDI-VC, além da maior complexidade dos problemas, os estudantes são estimulados a utilizar a biblioteca didática `morph.py`, o que introduz uma camada adicional de abstração e de conhecimento específico do domínio.

Esse padrão é relevante para a discussão pedagógica, mas **não permite atribuir as avaliações inferiores à utilização da IA**. Os recursos também diferem quanto ao objetivo pedagógico, à forma de interação, ao grau de autonomia proporcionado ao estudante, à complexidade das tarefas e ao momento de utilização. Portanto, as diferenças observadas devem ser interpretadas como características da avaliação dos recursos em seus respectivos contextos pedagógicos, e não como evidência isolada de um efeito da utilização da IA.

### Comparações múltiplas — Pós-teste de Conover 

Como o teste de Friedman foi significativo, foi aplicado o pós-teste de
*Conover* [@conover1971], com correção de *Bonferroni* [@dunn1961],
para identificar diferenças entre pares de recursos.

O resultado geral indica que o **Material Didático por IA (`Q08_02`)**, gerado pelo *Gemini Notebook*, se encontra entre os recursos com menor avaliação e apresenta diferenças significativas em relação a parte dos recursos mais bem avaliados.

Os resultados do pós-teste devem ser interpretados considerando a correção para comparações múltiplas. Portanto, **diferenças descritivas entre os recursos não devem ser interpretadas automaticamente como diferenças estatisticamente significativas**.

A análise completa do pós-teste é apresentada graficamente e em tabelas no *notebook* estatístico que acompanha este apêndice. O material será disponibilizado após a anonimização dos dados.

## *Ranking* agregado por frequência de seleção dos recursos didáticos — Q09 

A questão `Q09` utilizou um formato de seleção de preferência, no qual cada estudante deveria **selecionar exatamente quatro recursos** considerados mais eficazes para seu processo de estudo, dentre as dez opções apresentadas. Não havia ordenação entre as quatro opções escolhidas. Dessa forma, Q09 não produz uma escala ordinal nem um *ranking* individual; produz **frequências agregadas de seleção**. O *ranking* apresentado na @tbl-ranking-q09 foi construído pela frequência de seleção de cada recurso.

| Posição | Recurso                                      | Escolhas | Percentual |
| :-----: | :-------------------------------------------- | -------: | ---------: |
|    1º   | Exercícios Práticos (EPs em aula)            |       62 |      78,5% |
|    2º   | Biblioteca `morph.py`                        |       58 |      73,4% |
|    3º   | Simuladores Interativos                      |       44 |      55,7% |
|    4º   | Livro Interativo                             |       43 |      54,4% |
|    5º   | Simulados no *Safe Exam Browser* (SEB)       |       40 |      50,6% |
|    6º   | `TestSuite.py`                               |       32 |      40,5% |
|    7º   | Casos de teste automatizados no VPL          |       21 |      26,6% |
|    8º   | *Gemini Notebook* — recursos de áudio/visual |        6 |       7,6% |
|    9º   | *Feedback* Socrático por IA no VPL           |        5 |       6,3% |
|   10º   | Rubrica de correção por IA                   |        5 |       6,3% |

: *Ranking* agregado de preferência declarada dos recursos didáticos ($N=79$), construído pela frequência de seleção. Cada estudante selecionou exatamente quatro recursos, sem estabelecer ordem entre eles. {#tbl-ranking-q09}

Diferentemente dos itens `Q08`, a questão `Q09` não solicitou uma avaliação independente de cada recurso em uma escala de 1 a 5. Cada estudante deveria selecionar exatamente quatro recursos, sem atribuir posições entre eles. Assim, os resultados de `Q09` representam preferência declarada por seleção, e não intensidade de avaliação ou ordenação ordinal individual.

Como participaram 79 estudantes e cada um selecionou exatamente quatro recursos, foram registradas 316 seleções no total ($79\times4$). A soma das frequências apresentadas na @tbl-ranking-q09 também é 316, confirmando a consistência da tabulação.

Os **EPs em aula** e a biblioteca **`morph.py`** foram os recursos mais escolhidos, enquanto os recursos associados diretamente à geração de conteúdo, *feedback* ou rubricas por IA ficaram entre os menos selecionados.

A convergência entre `Q08` e `Q09` reforça a importância atribuída pelos estudantes à **prática guiada, à programação e à interação direta com os recursos da disciplina**.

## Avaliação Global dos Recursos Didáticos — Q10 

A questão `Q10` (`Q10-GLOBAL_BLOCO3`) fornece uma avaliação global dos recursos didáticos apresentados no Bloco 3.

A avaliação global apresentou média de aproximadamente **4,08**, com mediana **4**, indicando percepção geral positiva dos recursos utilizados na disciplina.

Para investigar a relação entre as avaliações específicas e a avaliação global,
foi calculado o **escore médio dos dez itens de `Q08`** e correlacionado com
`Q10` por meio do coeficiente de *Spearman* [@spearman1904]:

$$
\rho = 0,520,\qquad p < 0,001.
$$

A correlação positiva e moderada indica que estudantes que avaliaram mais favoravelmente o conjunto dos recursos também tenderam a apresentar avaliações globais mais positivas.

Esse resultado demonstra coerência entre as avaliações específicas e a percepção global, mas não permite determinar quais recursos individualmente explicam a avaliação de `Q10`.

In [ ]:
# ── Correlação de Spearman (Q08 médio × Q10) e alfa de Cronbach (Q08) ────
if DADOS_DIR is not None:
    escore_q08 = pos[Q08_ITENS].mean(axis=1)
    rho, p_rho = stats.spearmanr(escore_q08, pos['Q10'], nan_policy='omit')
    print(f'Spearman (Q08 médio × Q10): ρ = {rho:.3f}, p = {p_rho:.3g}')

    def cronbach_alpha(df: pd.DataFrame) -> float:
        itens = df.dropna()
        k = itens.shape[1]
        var_itens = itens.var(axis=0, ddof=1).sum()
        var_total = itens.sum(axis=1).var(ddof=1)
        return (k / (k - 1)) * (1 - var_itens / var_total)

    alpha_q08 = cronbach_alpha(pos[Q08_ITENS])
    print(f'alfa de Cronbach (Q08_01..Q08_10) = {alpha_q08:.3f}')

    alpha_q11_q12 = cronbach_alpha(pos[['Q11', 'Q12']])
    print(f'alfa de Cronbach (Q11, Q12) = {alpha_q11_q12:.3f}')

    rho2, p_rho2 = stats.spearmanr(
        pos[['Q11', 'Q12']].mean(axis=1), pos['Q13'], nan_policy='omit'
    )
    print(f'Spearman ((Q11+Q12)/2 × Q13): ρ = {rho2:.3f}, p = {p_rho2:.3g}')


## Confiabilidade dos Recursos Didáticos 

A consistência interna dos dez itens de `Q08` foi avaliada pelo $\alpha$ de
Cronbach [@cronbach1951], obtendo-se o valor:

$$
\alpha = 0,847.
$$

O valor $\alpha=0,847$ indica **boa consistência interna** entre os dez itens, segundo critérios convencionais de interpretação. Entretanto, os dez itens representam recursos pedagogicamente distintos. Portanto, uma consistência interna adequada não significa que todos constituam uma única intervenção ou construto pedagógico. O resultado indica apenas associação suficiente entre os padrões de resposta para justificar a análise conjunta dos itens, sem implicar unidimensionalidade da escala.

Entretanto, os dez itens representam recursos pedagogicamente distintos.
Portanto, uma consistência interna adequada não significa que todos
constituam uma única intervenção ou construto pedagógico. O resultado indica
apenas coerência suficiente entre as respostas para a análise conjunta da
escala.

## Diferenças entre Turmas 

As avaliações dos recursos didáticos também foram comparadas entre as diferentes turmas, compostas por **duas turmas da graduação e uma turma do mestrado**.

Para a avaliação global `Q10`, o teste de *Kruskal--Wallis* apresentou:

$$
H = 4,608,\qquad p = 0,100.
$$

Não foi observada diferença estatisticamente significativa entre as turmas ao nível de 5%. Assim, embora possam existir diferenças descritivas entre os grupos, não há evidência estatística suficiente para afirmar que as turmas avaliaram globalmente os recursos didáticos de maneira diferente.

In [ ]:
# ── Kruskal-Wallis: diferenças entre turmas (DA1, DA2, MCC) ──────────────
if DADOS_DIR is not None:
    grupos_q10 = [g['Q10'].dropna() for _, g in pos.groupby('turma')]
    h_q10, p_q10 = stats.kruskal(*grupos_q10)
    print(f'Kruskal-Wallis Q10 por turma: H = {h_q10:.3f}, p = {p_q10:.3g}')

    grupos_q13 = [g['Q13'].dropna() for _, g in pos.groupby('turma')]
    h_q13, p_q13 = stats.kruskal(*grupos_q13)
    print(f'Kruskal-Wallis Q13 por turma: H = {h_q13:.3f}, p = {p_q13:.3g}')


## Uso Transversal e Integração da IA 

Enquanto o Bloco 3 avalia recursos didáticos específicos, o Bloco 4 investiga diretamente a percepção dos estudantes sobre a **integração da IA no processo de ensino e aprendizagem**.

Foram analisados três itens:

* `Q11` — Pensamento crítico e alucinações;
* `Q12` — Transversalidade pedagógica, aplicada em outras disciplinas;
* `Q13` — Avaliação geral da integração de IA.

As estatísticas descritivas e os respectivos testes de diferença em relação ao ponto neutro da escala são apresentados na @tbl-ia-transversal.

| Item  | Descrição                           | Média | Mediana | $p$ vs. neutro |
| :---- | :---------------------------------- | ----: | ------: | -------------: |
| `Q11` | Pensamento crítico e alucinações    |  3,47 |       4 |       $<0,001$ |
| `Q12` | Transversalidade pedagógica         |  3,51 |       4 |          0,002 |
| `Q13` | Avaliação geral da integração de IA |  3,63 |       4 |       $<0,001$ |

: Avaliação dos itens sobre o uso transversal e a integração da IA no processo de ensino e aprendizagem ($N=79$). Os valores de $p$ referem-se ao teste de postos sinalizados de *Wilcoxon* para uma amostra, tendo 3,0 como valor de referência. {#tbl-ia-transversal}

Os três itens apresentaram **evidência estatística de avaliações superiores ao ponto neutro da escala (3,0)**. A maior média foi observada em `Q13` (3,63), referente à avaliação geral da integração da IA, seguida por `Q12` (3,51) e `Q11` (3,47). As medianas dos três itens foram iguais a 4. Esses resultados indicam uma **percepção globalmente positiva dos estudantes sobre a integração da IA** no processo de ensino e aprendizagem, embora com avaliações mais moderadas do que as observadas para a maioria dos recursos didáticos tradicionais ou diretamente relacionados às atividades práticas.

A consistência interna dos itens `Q11` e `Q12` foi avaliada pelo $\alpha$ de *Cronbach* [@cronbach1951]:

$$
\alpha = 0,780.
$$

Esse resultado indica associação interna adequada entre os dois itens. Entretanto, o coeficiente deve ser interpretado com cautela, pois foi calculado com apenas dois itens e, nesse caso, está diretamente relacionado à correlação entre eles.

A relação entre os aspectos de pensamento crítico e transversalidade (`Q11` e `Q12`) e a avaliação global da integração da IA (`Q13`) foi forte e estatisticamente significativa:

$$
\rho = 0,774,\qquad p < 0,001.
$$

Assim, estudantes com percepção mais favorável sobre o uso crítico e transversal da IA tendem também a avaliar mais positivamente sua integração geral na disciplina.

Para `Q13`, a comparação entre as turmas não apresentou diferença estatisticamente significativa:

$$
H = 3,113,\qquad p = 0,211.
$$

## Síntese dos Resultados 

A @fig-boxplot-sintese-likert apresenta a distribuição das respostas dos **21 itens *Likert*** analisados no questionário PÓS. A ordenação segue a média decrescente, permitindo visualizar conjuntamente a posição central, a dispersão e a relação das respostas com o ponto neutro da escala.

Os resultados convergem para **três padrões principais**: a valorização de atividades práticas e interativas; a ausência de evidência estatística de mudança individual nas seis dimensões comparadas entre PRÉ e PÓS; e uma avaliação globalmente positiva da integração da IA, embora os recursos nos quais a IA participou diretamente da geração de conteúdo, *feedback* ou rubricas tenham recebido avaliações inferiores às de vários recursos práticos.

::: {#fig-boxplot-sintese-likert}
![](imagens/apendice_f_boxplot.png){fig-align="center" width="100%"}

Distribuição das respostas dos 21 itens *Likert* do questionário PÓS, ordenados pela média decrescente. O Material Didático por IA, gerado com o *Gemini Notebook*, apresentou a pior média entre os itens analisados.
:::

A análise longitudinal deve ser interpretada com cautela, pois somente **10 estudantes** puderam ser identificados nos dois momentos. Nesse grupo, não foi encontrada evidência estatisticamente significativa de mudança em nenhuma das seis dimensões analisadas. As diferenças observadas nas amostras completas pelo teste de *Mann--Whitney U* devem ser interpretadas como diferenças entre os conjuntos de respondentes, e não como evidência de evolução individual.

Quanto aos recursos didáticos, o teste de Friedman mostrou que as dez avaliações não foram estatisticamente equivalentes ($\chi^2=189,299$, $p<0,001$), com magnitude moderada segundo o W de Kendall ($W=0,266$). Os resultados de `Q08` e `Q09` convergem ao destacar **prática guiada, programação, experimentação e interação** como componentes particularmente valorizados.

A integração da IA, por sua vez, recebeu avaliação global positiva. Entretanto, os três recursos com participação direta da IA na geração de conteúdo, *feedback* ou rubricas apresentaram médias inferiores às de diversos recursos práticos e interativos. O **Material Didático por IA**, produzido com o *Gemini Notebook*, apresentou a menor média entre os itens de `Q08` (3,08) e foi o único dos 21 itens analisados cuja avaliação **não apresentou evidência estatística de diferença em relação ao ponto neutro** da escala.

Esse padrão é **compatível com a hipótese de que a percepção dos estudantes dependa não apenas da presença da IA, mas também da forma como ela é integrada à atividade pedagógica**. Trata-se, contudo, de uma interpretação exploratória, que não permite estabelecer efeito causal da forma de integração da IA sobre as avaliações.

## Tabela Síntese dos Resultados Estatísticos 

Como síntese quantitativa final, a @tbl-sintese-geral reúne os **21 itens *Likert*** analisados nos Blocos 2, 3 e 4 do questionário PÓS. Esses itens correspondem a `Q01`–`Q07`, `Q08_01`–`Q08_10`, `Q10` e `Q11`–`Q13`. A tabela está ordenada pela média decrescente, permitindo comparar diretamente as diferentes dimensões avaliadas pelos estudantes.

Para cada item são apresentados o bloco, a identificação da questão, uma descrição resumida, a média, o desvio-padrão (DP), a mediana e o valor de $p$ obtido pelo **teste de postos sinalizados de *Wilcoxon* para uma amostra**, tendo como referência o ponto neutro da escala (3,0). O número de respostas válidas é constante ($N=79$) e, por isso, é informado no *caption* da tabela.

O teste de *Wilcoxon* foi utilizado para verificar se as respostas apresentavam evidência de deslocamento em relação ao ponto neutro da escala. Adotou-se o nível de significância de 5%, considerando-se estatisticamente significativo o resultado com $p<0,05$. A interpretação da direção considera o conjunto das estatísticas descritivas e o sentido do deslocamento: valores superiores a 3,0 são interpretados como avaliações favoráveis em relação ao ponto neutro.

Dos **21 itens *Likert*** analisados, **20 (95,2%) apresentaram evidência estatística de deslocamento favorável em relação ao ponto neutro**. Apenas o `Q08_02` — Material Didático por IA — não apresentou diferença estatisticamente significativa em relação a 3,0 ($p=0,758$).

: Síntese estatística dos itens *Likert* dos Blocos 2, 3 e 4, ordenada pela média decrescente ($N=79$). Os valores de $p$ referem-se ao teste de postos sinalizados de *Wilcoxon* para uma amostra, tendo 3,0 como valor de referência. {#tbl-sintese-geral}

| Bloco                       | Item     | Descrição resumida                     | Média |   DP | Mediana | $p$ vs. 3,0 | Sig.? |
| :-------------------------- | :------- | :------------------------------------- | ----: | ---: | ------: | ----------: | :---: |
| Recursos Didáticos          | `Q08_04` | Resolução de EPs em aula               |  4,59 | 0,65 |     5 |    $<0,001$ |  Sim  |
| Competências                | `Q01`    | Capacidade de programação              |  4,58 | 0,74 |     5 |    $<0,001$ |  Sim  |
| Recursos Didáticos          | `Q08_03` | Simuladores interativos                |  4,35 | 0,85 |     5 |    $<0,001$ |  Sim  |
| Recursos Didáticos          | `Q08_01` | Livro interativo                       |  4,30 | 0,81 |     4 |    $<0,001$ |  Sim  |
| Recursos Didáticos          | `Q08_05` | Biblioteca `morph.py`                  |  4,29 | 1,05 |     5 |    $<0,001$ |  Sim  |
| Recursos Didáticos          | `Q08_10` | Uso do `TestSuite.py`                  |  4,25 | 0,94 |     5 |    $<0,001$ |  Sim  |
| Recursos Didáticos          | `Q08_09` | Correção objetiva no VPL               |  4,20 | 1,02 |     5 |    $<0,001$ |  Sim  |
| Global — Competências       | `Q07`    | Avaliação geral do aprendizado         |  4,11 | 0,89 |     4 |    $<0,001$ |  Sim  |
| Recursos Didáticos          | `Q08_06` | Provas e simulados paramétricos        |  4,09 | 1,08 |     4 |    $<0,001$ |  Sim  |
| Global — Recursos Didáticos | `Q10`    | Avaliação geral dos recursos didáticos |  4,08 | 0,83 |     4 |    $<0,001$ |  Sim  |
| Competências                | `Q05`    | Autonomia com bibliotecas              |  3,82 | 0,93 |     4 |    $<0,001$ |  Sim  |
| Competências                | `Q03`    | Atuação profissional                   |  3,75 | 1,03 |     4 |    $<0,001$ |  Sim  |
| Competências                | `Q02`    | Atuação acadêmica                      |  3,71 | 1,01 |     4 |    $<0,001$ |  Sim  |
| IA Transversal              | `Q13`    | Avaliação geral da integração de IA    |  3,63 | 1,24 |     4 |    $<0,001$ |  Sim  |
| Recursos Didáticos          | `Q08_08` | Rubrica por IA em simulados/provas     |  3,57 | 1,09 |     3 |    $<0,001$ |  Sim  |
| Competências                | `Q04`    | Fundamentos matemáticos                |  3,56 | 1,03 |     4 |    $<0,001$ |  Sim  |
| IA Transversal              | `Q12`    | Transversalidade pedagógica            |  3,51 | 1,21 |     4 |       0,002 |  Sim  |
| IA Transversal              | `Q11`    | Pensamento crítico e alucinações       |  3,47 | 1,13 |     4 |    $<0,001$ |  Sim  |
| Competências                | `Q06`    | *Feedback* por IA                      |  3,44 | 1,34 |     4 |       0,007 |  Sim  |
| Recursos Didáticos          | `Q08_07` | *Feedback* socrático no VPL            |  3,42 | 1,23 |     4 |       0,006 |  Sim  |
| Recursos Didáticos          | `Q08_02` | Material didático por IA               |  3,08 | 1,37 |     3 |       0,758 |  Não  |

A ordenação evidencia que os itens com maiores médias estão associados principalmente à **capacidade de programação, à resolução prática de exercícios e aos recursos interativos**. A menor média foi observada no **Material Didático por IA** (`Q08_02`), único item cuja avaliação não apresentou evidência estatística de diferença em relação ao ponto neutro da escala.

Esse resultado reforça a distinção apresentada anteriormente entre a **avaliação dos recursos didáticos** e a **avaliação da integração da IA**. Embora a integração da IA tenha recebido avaliação global positiva, os recursos nos quais a IA participou diretamente da produção de conteúdo, *feedback* ou rubricas apresentaram avaliações mais moderadas. Essa diferença, contudo, **não deve ser interpretada como evidência de efeito negativo da IA**, pois os recursos avaliados possuem características pedagógicas distintas.

## Principais Conclusões 

Os resultados permitem destacar as seguintes conclusões:

1. **A experiência pedagógica foi avaliada positivamente.** A avaliação global dos recursos didáticos (`Q10`) apresentou média de aproximadamente **4,08** em uma escala de cinco pontos e mediana igual a 4.

2. **Os recursos didáticos foram avaliados de maneira estatisticamente diferente.** O teste de Friedman foi significativo ($\chi^2=189,299$, $p<0,001$), com magnitude moderada segundo o W de Kendall ($W=0,266$).

3. **A prática guiada foi o componente mais valorizado.** A Resolução de EPs em Aula apresentou a maior média em `Q08` (4,59) e foi o recurso mais escolhido em `Q09` (78,5%).

4. **`morph.py` e os simuladores também se destacaram.** Ambos aparecem entre os recursos mais bem avaliados e reforçam a importância da experimentação e da programação no processo de aprendizagem.

5. **Não foi encontrada evidência estatisticamente significativa de mudança individual entre PRÉ e PÓS.** Entre os **10 estudantes identificados nos dois momentos**, nenhuma das seis dimensões do Bloco 2 apresentou diferença estatisticamente significativa no teste de *Wilcoxon*. Esse resultado deve ser interpretado considerando o reduzido tamanho da amostra pareada e o baixo poder estatístico potencial para detectar mudanças de pequena ou moderada magnitude.

6. **A integração da IA foi avaliada positivamente, mas de maneira não uniforme entre os recursos.** Os três itens do Bloco 4 apresentaram evidência estatística de avaliações superiores ao ponto neutro, enquanto os recursos com participação direta da IA na geração de conteúdo, *feedback* ou rubricas apresentaram médias inferiores às de diversos recursos práticos.

7. **As avaliações específicas estão associadas às avaliações globais.** A correlação entre o escore médio de `Q08` e `Q10` foi moderada e significativa ($\rho=0,520$, $p<0,001$), enquanto a associação entre `Q11`/`Q12` e `Q13` foi forte e significativa ($\rho=0,774$, $p<0,001$).

8. **Não foram observadas diferenças globais significativas entre as turmas.** Para `Q10`, o teste de *Kruskal--Wallis* resultou em $p=0,100$; para `Q13`, em $p=0,211$.

9. **A instrumentação apresentou consistência interna adequada.** O conjunto de recursos didáticos apresentou $\alpha=0,847$, enquanto os dois itens relacionados ao pensamento crítico e à transversalidade da IA apresentaram $\alpha=0,780$. Este último resultado deve ser interpretado com cautela devido ao número reduzido de itens.

Em conjunto, os resultados sugerem que a avaliação positiva da experiência está mais associada à **forma como os recursos são incorporados ao processo de aprendizagem** do que simplesmente à presença de tecnologia ou IA. As atividades práticas, os simuladores, a biblioteca `morph.py` e os instrumentos de apoio à programação foram particularmente valorizados.

Ao mesmo tempo, os recursos nos quais a IA participou diretamente da produção de conteúdo, *feedback* ou rubricas apresentaram avaliações inferiores às de diversos recursos práticos. Esse padrão **não demonstra que a utilização da IA tenha causado avaliações menos favoráveis**. Os recursos diferem quanto à finalidade, ao formato, à interação, ao momento de aplicação, ao grau de autonomia proporcionado ao estudante e à participação docente e discente.

A análise PRÉ--PÓS também não forneceu evidência de mudança individual estatisticamente significativa nas seis dimensões avaliadas. A interpretação desse resultado deve considerar, particularmente, o reduzido número de estudantes identificados nos dois momentos ($N=10$), que limita a capacidade de detectar mudanças.

> **Nota metodológica.** Os resultados estatísticos devem ser interpretados como evidências de **diferenças, associações e padrões de percepção**, e não como demonstrações de causalidade. Em particular, a menor avaliação dos recursos com participação direta da IA não permite concluir que a IA tenha causado essa percepção. Da mesma forma, as diferenças observadas entre as amostras completas PRÉ e PÓS não constituem evidência de evolução individual. A interpretação pedagógica deve considerar conjuntamente **significância estatística, tamanho de efeito, *ranking* agregado por frequência de seleção, correlações, consistência interna, composição da amostra e características concretas de cada recurso**.

## Avaliação de Valência das Respostas Abertas — Q14 e Q15 

As questões abertas `Q14` e `Q15` complementam as avaliações estruturadas do questionário PÓS, permitindo identificar aspectos da experiência pedagógica mencionados espontaneamente pelos estudantes. A **análise temática** apresentada anteriormente identifica *quais* aspectos foram mencionados; a análise de **valência** acrescenta *como* esses aspectos foram avaliados.

As respostas foram classificadas como **positivas, negativas, mistas ou neutras**. Primeiro, foi atribuída uma valência à resposta completa; posteriormente, quando necessário, realizou-se um refinamento por tema para distinguir avaliações diferentes dentro de uma mesma resposta.

Como `Q14` e `Q15` possuem finalidades distintas — respectivamente, identificar ganhos de aprendizagem e dificuldades ou necessidades de ajuste —, seus resultados de valência são interpretados separadamente, e não como medidas diretamente comparáveis de satisfação geral.

### Unidade de análise e critérios 

A unidade primária da análise foi a **resposta completa de cada participante**. Assim, cada resposta recebeu inicialmente uma única classificação de valência, evitando que respostas mais longas ou que mencionassem diversos aspectos tivessem maior peso na distribuição global.

Foram identificadas **40 respostas não vazias em `Q14`** e **38 em `Q15`**, correspondendo, respectivamente, a **50,6%** e **48,1%** dos 79 respondentes do questionário PÓS. As células vazias foram excluídas da análise.

Os critérios utilizados são apresentados na @tbl-criterios-valencia.

| Código       | Critério                                                                                                    |
| ------------ | ----------------------------------------------------------------------------------------------------------- |
| **POSITIVA** | Predominam benefício, aprovação ou melhoria percebida.                                                      |
| **NEGATIVA** | Predominam crítica, dificuldade ou ausência de benefício.                                                   |
| **MISTA**    | Aspectos positivos e negativos relevantes coexistem explicitamente.                                         |
| **NEUTRA**   | A resposta apenas descreve ou identifica um elemento, sem avaliação suficiente para estabelecer polaridade. |

: Critérios de classificação da valência. {#tbl-criterios-valencia}

A análise temática utilizou **codificação múltipla**, de modo que uma mesma resposta pôde receber mais de um tema. Foram identificadas **44 menções temáticas em `Q14`** e **59 em `Q15`**. Portanto, os totais temáticos não correspondem necessariamente ao número de respostas e não devem ser confundidos com os resultados da análise global de valência.

### Resultados globais por resposta 

A @tbl-valencia apresenta a distribuição da valência considerando cada resposta completa como uma única unidade de análise. Os percentuais foram calculados sobre as respostas não vazias de cada questão.

| Questão |   Positiva |   Negativa |     Mista |   Neutra | Total |
| ------- | ---------: | ---------: | --------: | -------: | ----: |
| `Q14`   | 32 (80,0%) |   3 (7,5%) |  2 (5,0%) | 3 (7,5%) |    40 |
| `Q15`   |   0 (0,0%) | 29 (76,3%) | 6 (15,8%) | 3 (7,9%) |    38 |

: Distribuição da valência global por resposta. {#tbl-valencia}

Em `Q14`, que solicita aos estudantes a indicação do uso da IA que proporcionou maior ganho para a aprendizagem, predominam respostas de valência positiva (80,0%). Ainda assim, foram registradas respostas negativas, mistas e neutras, indicando que os diferentes usos da IA não foram avaliados de forma uniformemente favorável.

Em `Q15`, que solicita a identificação de dificuldades ou de aspectos que necessitariam de ajustes em futuras ofertas, predominam respostas de valência negativa (76,3%). Nesse caso, a classificação negativa expressa críticas, limitações ou necessidades de melhoria em elementos específicos e **não deve ser interpretada como uma avaliação global negativa da disciplina ou da intervenção pedagógica**.

As distribuições refletem, portanto, as finalidades distintas das duas perguntas e não constituem medidas diretamente comparáveis de satisfação ou avaliação global.

### Refinamento por tema 

As **8 respostas classificadas globalmente como mistas** — `Q14`: 12 e 18; `Q15`: 4, 12, 15, 22, 29 e 37 — foram relidas para identificar a valência associada especificamente a cada tema mencionado. Esse procedimento foi necessário porque uma mesma resposta poderia, por exemplo, elogiar um recurso e criticar outro.

Nas respostas não mistas, a valência global foi mantida para os temas identificados. O refinamento resultou na alteração de **14 pares `(resposta, tema)`**.

Diferentemente da análise anterior, os totais das @tbl-valencia-tema-q14 e @tbl-valencia-tema-q15 correspondem a **menções temáticas**, pois uma mesma resposta pode contribuir para mais de uma categoria.

| Categoria                        | Positiva | Negativa | Mista | Neutra | Total |
| -------------------------------- | -------: | -------: | ----: | -----: | ----: |
| Geração de material              |       22 |        0 |     1 |      0 |    23 |
| EPs/prática em geral             |        6 |        0 |     0 |      1 |     7 |
| *Feedback* de IA nos EPs         |        4 |        0 |     0 |      2 |     6 |
| *Feedback*/rubrica em provas     |        3 |        0 |     0 |      1 |     4 |
| Nenhum ganho / rejeita uso de IA |        0 |        2 |     0 |      0 |     2 |
| Não classificável                |        1 |        1 |     0 |      0 |     2 |

: Valência por categoria temática — `Q14`. {#tbl-valencia-tema-q14}

| Categoria                      | Positiva | Negativa | Mista | Neutra | Total |
| ------------------------------ | -------: | -------: | ----: | -----: | ----: |
| *Feedback* por IA              |        0 |       12 |     0 |      0 |    12 |
| VPL / restrição de rede        |        0 |       11 |     0 |      0 |    11 |
| Biblioteca (`morph.py`/OpenCV) |        0 |        7 |     1 |      1 |     9 |
| SEB                            |        0 |        5 |     3 |      0 |     8 |
| Material gerado por IA         |        0 |        5 |     1 |      0 |     6 |
| Tempo de prova/simulado        |        0 |        5 |     0 |      0 |     5 |
| Outro                          |        0 |        4 |     0 |      0 |     4 |
| MCTest                         |        0 |        2 |     0 |      1 |     3 |
| Nenhuma dificuldade relevante  |        0 |        0 |     0 |      1 |     1 |

: Valência por categoria temática — `Q15`. {#tbl-valencia-tema-q15}

Em `Q14`, a categoria **geração de material** concentra 22 menções positivas e uma mista, constituindo o principal tema associado aos ganhos de aprendizagem. Os demais temas também apresentam predominância positiva ou neutra. As duas menções a **nenhum ganho / rejeita uso de IA** foram classificadas como negativas.

Em `Q15`, predominam avaliações negativas em praticamente todas as categorias. Destacam-se **feedback por IA**, com 12 menções negativas, e **VPL / restrição de rede**, com 11. A categoria **biblioteca (`morph.py`/OpenCV)** reúne 7 menções negativas, uma mista e uma neutra, enquanto o **SEB** apresenta 5 negativas e 3 mistas.

A concentração de valências mistas em `Q15` ocorre principalmente no **SEB**, responsável por 3 das 4 menções mistas da questão. Isso sugere que críticas ao ambiente seguro coexistiram, em algumas respostas, com o reconhecimento de sua necessidade ou função nas avaliações. Em contraste, *feedback* por IA, VPL/restrições de rede e tempo de prova ou simulado aparecem exclusivamente associados a avaliações negativas na codificação temática.

Em `Q14`, permanece apenas uma menção mista, relacionada à **geração de material**. O refinamento permitiu preservar avaliações distintas presentes na mesma resposta, sem atribuir indistintamente a valência global mista a todos os temas mencionados.

### Observações metodológicas 

* As categorias **POSITIVA**, **NEGATIVA**, **MISTA** e **NEUTRA** foram mantidas separadas. Não foram agregadas respostas ou menções mistas às polaridades positiva ou negativa, pois não foi definida uma regra adicional para essa agregação.

* O refinamento por tema foi aplicado às respostas classificadas como **mistas**. Nas demais, a valência global foi mantida para os temas identificados. Essa decisão evita uma segmentação sistemática em unidades menores de sentido, que aumentaria a complexidade da codificação e poderia atribuir maior peso a respostas mais longas.

* A classificação global e o refinamento por tema foram realizados manualmente e possuem caráter interpretativo. Não houve segundo codificador independente nem cálculo de concordância intercodificadores, como o *kappa* de Cohen. Os resultados devem, portanto, ser interpretados como uma **codificação analítica documentada**.

* As frequências possuem finalidade **descritiva**. Considerando o número de respostas não vazias e a baixa frequência de algumas categorias, os resultados descrevem os padrões observados no conjunto analisado, sem pretensão de extrapolação estatística para a população de estudantes.

### Auditabilidade 

A análise foi estruturada para permitir a rastreabilidade entre as respostas originais, a codificação individual e as tabelas agregadas. O arquivo `respostas_codificadas_final.csv` registra cada par `(questão, resposta, tema)`, incluindo a valência global da resposta, a valência refinada do tema e a indicação de eventual ajuste.

As tabelas agregadas correspondem aos arquivos `valencia_por_categoria_q14.csv` e `valencia_por_categoria_q15.csv`, produzidos pelo script `tabulacao_valencia_tematica.py` a partir de `respostas_q14_q15.csv`. Dessa forma, é possível reproduzir as frequências apresentadas nas @tbl-valencia-tema-q14 e @tbl-valencia-tema-q15.

In [ ]:
# ── Tabulação de valência das respostas abertas Q14 e Q15 ────────────────
# Reproduz respostas_codificadas_final.csv → valencia_por_categoria_q14.csv
# e valencia_por_categoria_q15.csv (ver tabulacao_valencia_tematica.py).

if DADOS_DIR is not None:
    codificadas = pd.read_csv(DADOS_DIR / 'respostas_codificadas_final.csv')

    # Valência global por resposta (uma linha por resposta não vazia)
    valencia_global = (
        codificadas.drop_duplicates(subset=['questao', 'resposta_id'])
        .groupby('questao')['valencia_global']
        .value_counts(normalize=False)
        .unstack(fill_value=0)
    )
    display(valencia_global)

    # Valência refinada por tema (uma linha por par resposta-tema — codificação múltipla)
    for questao in ['Q14', 'Q15']:
        tbl = (
            codificadas[codificadas['questao'] == questao]
            .groupby('tema')['valencia_refinada']
            .value_counts()
            .unstack(fill_value=0)
        )
        tbl['Total'] = tbl.sum(axis=1)
        print(f'\nValência por categoria temática — {questao}')
        display(tbl.sort_values('Total', ascending=False))
